trees/btree/btree.go

```diff
@@ -95,6 +96,109 @@ func (tree *Tree) Get(key interface{}) (value interface{}, found bool) {
 	return nil, false
 }
 
+// GetClosest does the same as above but if the given key does not exists,
+// it tries to find the elements next to it.
+// If the given key exists, it returns twice the value and found is true.
+// If the given key is between tow other keys the previous value will be first and the next at the second place.
+// If the given key is less than the first key, it returns prev as nil and next as the first value.
+// If the given key is higher than the last key, it returns prev as the last value and next as nil.
+// If empty it returns nil, nil, false.
+func (tree *Tree) GetClosest(key interface{}) (prev, next interface{}, found bool) {
```

Comment:
returning 3 params is a bit hard to handle for consumers of gods. i'd rather split this function into Ceiling and Floor which has also been added to RedBlackTree/TreeMap. we could add a GetClosest that would simply return two parameters (key-value pair) of the closest node be it ceiling or floow... otherwise nil if the tree is empty.